### ISSUE TITLE
allow passing command line flags to haddock

### ISSUE THREAD

**Issue by ghorn** [tweag/rules_haskell#425]
I would like to be able to pass `--ignore-all-exports` to haddock in order to display haddock comments on internal functions. Could a field be added to `haskell_doc` for passing additional flags? Or maybe just a boolean field for this flag if you're afraid of users trying to unsafely override flags.

**Comment by ghorn** [comment]
from https://www.haskell.org/haddock/doc/html/invoking.html:

> --ignore-all-exports
> Causes Haddock to behave as if every module has the ignore-exports attribute ([Section 3.7, “Module Attributes”](https://www.haskell.org/haddock/doc/html/module-attributes.html)). This might be useful for generating implementation documentation rather than interface documentation, for example.

**Comment by guibou** [comment]
A bit of followup on that one.

This is a real pain. It is possible to set attributes to a rule which will be passed to the aspects (used by `haskell_doc`). However these attributes are limited: they must be a `string` from a finite list of possible values:

> Public aspect attributes are of type string and are called parameters. Parameters must have avalues attribute specified on them. In this case we have a parameter called extension that is allowed to have ‘*’, ‘h’, or ‘cc’ as a value.

(from https://docs.bazel.build/versions/master/skylark/aspects.html)

That's by *design* in bazel.

PR #444 was a PR with a proof of concept of a limited set of arguments. It was closed because the solution was a HACK.

**Comment by mboes** [comment]
Yet another reason to change the design in the following way?

```
$ bazel build //some/target@doc
```

i.e. use implicit targets for API documentation, like we currently do for REPL's?
